app/mixins/ui-kit/border.js

```diff
@@ -0,0 +1,82 @@
+import Mixin from '@ember/object/mixin';
+import { computed } from '@ember/object';
+import { or, none, not } from '@ember/object/computed';
+import { checkDictionary } from 'travis/utils/ui-kit/assertions';
+
+// Border
+const BORDER_WIDTHS = {
+  NONE: 'none',
+  ONE: '1',
+};
+
+export default Mixin.create({
+  // Public interface //
+  borderColor: null,
+  borderWidth: null,
+
+  // Private //
+  get blank() {
+    return '';
+  },
+
+  hasNoBorderColor: none('borderColor'),
+  hasBorderColor: not('hasNoBorderColor'),
+  borderColorClass: computed('hasNoBorderColor', 'borderColor', function () {
+    return this.hasNoBorderColor ? '' : `border-${this.borderColor}`;
+  }),
+
+  borderTopWidth: or('borderWidth.top', 'borderWidth.y', 'borderWidth.all', 'borderWidth', 'blank'),
+  borderRightWidth: or('borderWidth.right', 'borderWidth.x', 'borderWidth.all', 'borderWidth', 'blank'),
+  borderBottomWidth: or('borderWidth.bottom', 'borderWidth.y', 'borderWidth.all', 'borderWidth', 'blank'),
+  borderLeftWidth: or('borderWidth.left', 'borderWidth.x', 'borderWidth.all', 'borderWidth', 'blank'),
+
+  borderTopWidthClass: computed('borderTopWidth', function () {
+    return this.borderTopWidth.length === 0 ? '' : `border-t-${this.borderTopWidth}`;
```

Comment:
`this.borderTopWidth.length === 0` could be just [`isEmpty`](https://api.emberjs.com/ember/3.12/functions/@ember%2Futils/isEmpty), or the opposite [`isPresent`](https://api.emberjs.com/ember/3.12/functions/@ember%2Futils/isPresent)

Reply:
Yes definitely. And this has been moved into the `prefix` function